java-frontend/src/main/java/org/sonar/java/se/ProgramPoint.java

```diff
@@ -40,10 +40,13 @@ private ProgramPoint(CFG.Block block, int i) {
   }
 
   ProgramPoint next() {
-    // pointing one element behind is allowed, it represents terminator of the CFG block
-    Preconditions.checkState(i + 1 <= block.elements().size() + 1,
-      "CFG Block has %s elements but PP at %s was requested", block.elements().size(), i + 1);
-    return new ProgramPoint(block, i + 1);
+    int blockSize = block.elements().size();
+    int nextPP = this.i + 1;
+    // pointing one element above block size is allowed, it represents terminator of the CFG block
+    // FIXME when RelationalSV is involved, we can point up to two elements behind, which is fishy
```

Comment:
after some investigation I removed FIXME, because in fact it's valid behavior. I moved invariant check to constructor and added a comment explaining it